packages/eas-cli/src/commands/project/onboarding.ts

```diff
@@ -0,0 +1,225 @@
+      : 'expo-default-template';
+    const initialTargetProjectDir: string = targetProjectDirInput ?? `./${githubRepositoryName}`;
+
+    Log.log(`👋 Welcome to Expo, ${actor.username}!`);
+    Log.log('🚀 We will continue your onboarding process in EAS CLI');
+    Log.log();
+    Log.log(
+      `🔎 Let's start by cloning ${
+        app.githubRepository
+          ? `your project (${githubUsername}/${githubRepositoryName})`
+          : `default expo template project (${githubUsername}/${githubRepositoryName})`
+      } from GitHub and installing dependencies.`
+    );
+    Log.log();
+    const shouldContinue = await confirmAsync({ message: 'Do you want to continue?' });
+    if (!shouldContinue) {
+      throw new Error("Aborting, run the command again once you're ready.");
+    }
+
+    const { targetProjectDir } = await runGitCloneAsync({
+      githubUsername,
+      githubRepositoryName,
+      targetProjectDir: initialTargetProjectDir,
+    });
+
+    const vcsClient = new GitClient(targetProjectDir);
+    await installDependenciesAsync({
+      projectDir: targetProjectDir,
+    });
+    await vcsClient.trackFileAsync('package-lock.json');
+
+    const shouldSetupCredentials =
+      actor.preferences.onboarding.deviceType === OnboardingDeviceType.Device &&
+      actor.preferences.onboarding.environment === OnboardingEnvironment.DevBuild;
+    if (shouldSetupCredentials) {
+      Log.log('🔑 Now we need to set up build credentials for your project:');
+      await new SetUpBuildCredentialsCommandAction(
+        actor,
+        graphqlClient,
+        vcsClient,
+        analytics,
+        await getPrivateExpoConfigWithProjectIdAsync({
+          projectDir: targetProjectDir,
+          graphqlClient,
+          actor,
+        }),
+        getDynamicPrivateProjectConfigGetter({
+          projectDir: targetProjectDir,
+          graphqlClient,
+          actor,
+        }),
+        platform,
+        'development',
+        targetProjectDir
+      ).runAsync();
+    }
+
+    if (await vcsClient.hasUncommittedChangesAsync()) {
+      Log.log(
+        '📦 We will now commit the changes made by the configuration process and push them to GitHub:'
+      );
+      Log.log();
+      Log.log('🔍 Checking for changes in the repository...');
+      await vcsClient.showChangedFilesAsync();
+      await reviewAndCommitChangesAsync(
+        vcsClient,
+        `[EAS onboarding] Install dependencies${
+          shouldSetupCredentials ? 'and set up build credentials' : ''
+        }`,
+        { nonInteractive: false }
+      );
+      Log.log('📤 Pushing changes to GitHub...');
+      await runGitPushAsync({
+        targetProjectDir,
+      });
+    }
+
+    await UserPreferencesMutation.markCliDoneInOnboardingUserPreferencesAsync(graphqlClient, {
+      appId: app.id,
+    });
+
+    Log.log();
+    Log.log('🎉 We finished configuring your project.');
+    Log.log('🚀 You can now go back to the website to continue.');
```

Comment:
Can we show a link here?

Reply:
@tchayen what will happen if we link to `expo.new` from here?

Reply:
We can link back to the page where user came from. It is:

```js
const appSlug = '...';
const appOwnerAccountId = '...';
const url = `/onboarding/develop/set-up-project-on-your-machine?project=${appSlug}&accountId=${appOwnerAccountId}`
```

Reply:
awesome, thanks 🙏